### PR TITLE
Add possibility to use Room Manager inside example app

### PR DESCRIPTION
**Pull request by mironiasty**
## Description

Add possibility to use Room Manager from inside example app. I'm adding tabbar navigator so user can choose to use Room Manager API to get all data required to connect with Jellyfish.

Update app icons (replace default app icon with Jellyfish logo).

Remove flipper from app configuration (I don't think it is required in example app)

Fix eslint issues in component library.

## Motivation and Context
We want to create demo that shows end to end process of joining chat using simplified method - Room Manager.

## How has this been tested?
It was tested on local devices.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

## Screenshots (if appropriate)
Screenshot 2024-04-17 at 16 19 40

### CODE REVIEW

File: example/components/Button.tsx [hunk 1]
```diff
@@ -43,14 +43,14 @@ const Button = ({
   onPress,
   title,
 }: StandardButtonProps) => {
-  const GetBackgroundColorStyle = (
-    type: StandardButtonTypeName,
-    disabled: boolean,
+  const getBackgroundColorStyle = (
+    buttonType: StandardButtonTypeName,
```
Comment (mironiasty) [inline example/components/Button.tsx:47]: These are changes required to fix eslint issues

File: example/screens/ConnectWithRoomManagerScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+async function getJellyFishServer(
+  roomManagerUrl: string,
+  roomName: string,
+  userName: string,
+) {
+  const url = roomManagerUrl.endsWith('/')
+    ? roomManagerUrl
+    : roomManagerUrl + '/';
+  const result = await fetch(
+    `${url}rooms/${roomName.trim()}/users/${userName.trim()}`,
```
Comment (balins) [inline example/screens/ConnectWithRoomManagerScreen.tsx:36]: Consider adding
```suggestion
    `${url}api/rooms/${roomName.trim()}/users/${userName.trim()}`,
```
I'd expect the user to provide just protocol + host + port

File: example/screens/ConnectWithRoomManagerScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  return {
+    jellyfishUrl: tokenData.url,
+    token: tokenData.token,
+  };
+}
+
+const ConnectScreen = ({navigation}: Props) => {
+  const {connect} = useJellyfishClient();
+  const [connectionError, setConnectionError] = useState<string | undefined>(
+    undefined,
```
Comment (balins) [inline example/screens/ConnectWithRoomManagerScreen.tsx:53]: nit: for values that are set explicitly, I'd rather expect `null` than `undefined`
```suggestion
  const [connectionError, setConnectionError] = useState<string | null>(
    null,
```

File: example/screens/ConnectWithRoomManagerScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+import {usePermissionCheck} from '../hooks/usePermissionCheck';
+
+type Props = CompositeScreenProps<
+  BottomTabScreenProps<TabParamList, 'ConnectWithRoomManager'>,
+  NativeStackScreenProps<AppRootStackParamList>
+>;
+
+const {URL_INPUT, TOKEN_INPUT, CONNECT_BUTTON} = connectScreenLabels;
+
+async function getJellyFishServer(
```
Comment (balins) [inline example/screens/ConnectWithRoomManagerScreen.tsx:27]: nit: consistency
```suggestion
async function getJellyfishServer(
```

File: example/screens/ConnectWithRoomManagerScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+      await connect(jellyfishUrl, token);
+
+      navigation.navigate('Preview');
+    } catch (e) {
+      const message =
+        'message' in (e as Error) ? (e as Error).message : 'Unknown error';
+      setConnectionError(message);
+    } finally {
+      setLoading(false);
```
Comment (balins) [inline example/screens/ConnectWithRoomManagerScreen.tsx:83]: When testing on Android, I've encountered a bug where if connecting to the room manager without `/api` appended to its URL, the client gets 404, no error message shows up and the button never becomes active again.
Screenshot 2024-04-18 at 16 20 07

File: example/screens/ConnectWithRoomManagerScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+          <Image
+            style={styles.logo}
+            source={require('../assets/jellyfish-logo.png')}
+            resizeMode="contain"
+          />
+          <TextInput
+            onChangeText={setRoomManagerUrl}
+            value={roomManagerUrl}
+            accessibilityLabel={URL_INPUT}
+            placeholder="Room Manager URL"
```
Comment (karkakol) [inline example/screens/ConnectWithRoomManagerScreen.tsx:103]: Imo somewhere should be specified how to obtain this url (eg in app's readme.) and its format  (eg `<protocol>://<ip_address>:<port>`

**Review by karkakol — CHANGES_REQUESTED** [review]

**Review by karkakol — COMMENTED** [review]
In `.github/actions/cache_apk_file/action.yml` and `.github/actions/restore_apk_file/action.yml` files add path to hooks directory. It is necessary for cache in github actions.

Before merging let me know when all comments will be resolved. I will run locally e2e tests on Iphone and android device.

File: example/screens/ConnectWithRoomManagerScreen.tsx (hunk 2 above)
Comment (mironiasty) [reply example/screens/ConnectWithRoomManagerScreen.tsx:36]: After recent API changes it indeed makes sense

File: example/screens/ConnectWithRoomManagerScreen.tsx (hunk 3 above)
Comment (mironiasty) [reply example/screens/ConnectWithRoomManagerScreen.tsx:53]: We had some interesting discussion about `null` vs `undefined` and decided to go for `undefined` in such cases. But seems like `null` is more common in this codebase indeed, so I will update.

File: example/screens/ConnectWithRoomManagerScreen.tsx (hunk 5 above)
Comment (mironiasty) [reply example/screens/ConnectWithRoomManagerScreen.tsx:83]: Indeed, there was a bug related with new version of Room Manger

File: example/navigators/AppNavigator.tsx [hunk 7]
```diff
@@ -1,31 +1,72 @@
...
 };
 
+export type TabParamList = {
+  ConnectWithToken: undefined;
+  ConnectWithRoomManager: undefined;
+};
+
+const tabBarIcon = (icon: string) => () =>
+  <Typo variant="body-big">{icon}</Typo>;
+
```
[the remark this answers is not in the document]
Comment (mironiasty) [reply example/navigators/AppNavigator.tsx:25]: I didn't notice that example app already has `MaterialCommunityIcons` -> updated to use it.

File: example/screens/ConnectWithRoomManagerScreen.tsx (hunk 6 above)
Comment (mironiasty) [reply example/screens/ConnectWithRoomManagerScreen.tsx:103]: This part will be added once we finish `js-server-sdk`

File: example/screens/ConnectWithRoomManagerScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+  const [roomManagerUrl, setRoomManagerUrl] = useState(
+    process.env.ROOM_MANAGER_URL ?? '',
+  );
+  const [roomName, setRoomName] = useState('');
+  const [userName, setUserName] = useState('');
+
+  usePermissionCheck();
+
+  const onTapConnectButton = async () => {
```
Comment (karkakol) [inline example/screens/ConnectWithRoomManagerScreen.tsx:62]: useCallback ?

**Review by karkakol — COMMENTED** [review]
I do not changes related to this [comment](https://github.com/jellyfish-dev/react-native-client-sdk/pull/45#pullrequestreview-2011571790).

**Review by balins — APPROVED** [review]

File: example/screens/ConnectWithRoomManagerScreen.tsx (hunk 8 above)
Comment (mironiasty) [reply example/screens/ConnectWithRoomManagerScreen.tsx:62]: It will add only small benefit here - re-rendering button is cheap.

**Review by karkakol — APPROVED** [review]

**Pull request merged**